src/main/java/com/example/onlinebookstore/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import com.example.onlinebookstore.mapper.CategoryMapper;
+import com.example.onlinebookstore.model.Category;
+import com.example.onlinebookstore.repository.category.CategoryRepository;
+import com.example.onlinebookstore.service.CategoryService;
+import jakarta.persistence.EntityNotFoundException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper categoryMapper;
+
+    @Override
+    public List<CategoryDto> findAll(Pageable pageable) {
+        return categoryRepository.findAll().stream()
+                .map(categoryMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CategoryDto getById(Long id) {
+        Category category = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id " + id));
+        return categoryMapper.toDto(category);
```

Comment:
use Optional#map to reduce code.
```suggestion
        return categoryRepository.findById(id).map(categoryMapper::toDto).orElseThrow(
                () -> new EntityNotFoundException("Can't find category by id " + id));
```